Rework handling of active tab

wutschel opened the pull request:
## Description
<!--- Detailed info for reviewers and developers -->
This PR resolves a long-standing concept problem, which also is expected to cause some sporadically reported crashes in via AppStore while using Global Search. 

#### Background
`choosedTab` represents the selected view in a menu, e.g. Albums, Artists or Genre. The variable is used to read key data, like JSON methods/parameters or sort options, from the menu structure array. The implementation for now did change this ivar while processing actions for Global Search items (like entering next menu level or processing action sheets). This could result in the current Global Search view (always `choosedTab = 0`) temporarily using wrong values of `choosedTab` just to process an item related action. In worst case this could end up in an out-of-bounds access crash.

#### Main Change 
Instead of changing the ivar `choosedTab`, when processing actions in Global Search, it is better to use a local variable  -- `activeTab` was introduced. This avoids race conditions and allows to remove special handling when coming back to Global Search from a child view or after processing an action in Global Search. This requires the implementation of `getActiveTab` which returns the targeted `activeTab` for a given item.

#### Further Improvements
After this main rework similar changes were done to `NowPlaying` and `ShowInfoViewController`, replacing ivar with local variables, following the same naming convention as in `DetailViewController`.

#### Crash Log
One of AppStore's crash logs points to an out-of-bounds access in `NSDictionary *parameters = [Utilities indexKeyedDictionaryFromArray:menuItem.mainParameters[choosedTab]];` which is likely to be caused by `choosedTab` having been altered for Global Search.
```
Last Exception Backtrace:
0   CoreFoundation                	0x183152180 __exceptionPreprocess + 228 (NSException.m:172)
1   libobjc.A.dylib               	0x18232a9f8 objc_exception_throw + 56 (objc-exception.mm:557)
2   CoreFoundation                	0x1830cbbec _CFThrowFormattedException + 112 (CFObject.m:1958)
3   CoreFoundation                	0x183050bec -[__NSArrayM objectAtIndexedSubscript:] + 220 (NSArrayM.m:291)
4   Kodi Remote                   	0x1009dbfc8 -[DetailViewController indexAndDisplayData] + 108 (DetailViewController.m:4993)
5   Kodi Remote                   	0x1009d2c98 __43-[DetailViewController actionSheetHandler:]_block_invoke_2 + 324 (DetailViewController.m:3648)
6   UIKitCore                     	0x1afc59d1c -[UIViewAnimationBlockDelegate _didEndBlockAnimation:finished:context:] + 752 (UIView.m:12852)
7   UIKitCore                     	0x1afc30a74 -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 312 (UIView.m:0)
8   UIKitCore                     	0x1afc31048 -[UIViewAnimationState animationDidStop:finished:] + 296 (UIView.m:2111)
9   QuartzCore                    	0x1876a23c8 CA::Layer::run_animation_callbacks(void*) + 284 (CALayer.mm:6680)
10  libdispatch.dylib             	0x182b907d4 _dispatch_client_callout + 16 (object.m:511)
11  libdispatch.dylib             	0x182b3e008 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1068 (inline_internal.h:2441)
12  CoreFoundation                	0x1830e3b20 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12 (CFRunLoop.c:1813)
13  CoreFoundation                	0x1830dea58 __CFRunLoopRun + 1924 (CFRunLoop.c:3113)
14  CoreFoundation                	0x1830ddfb4 CFRunLoopRunSpecific + 436 (CFRunLoop.c:3247)
15  GraphicsServices              	0x1852e079c GSEventRunModal + 104 (GSEvent.c:2245)
16  UIKitCore                     	0x1af7cec38 UIApplicationMain + 212 (UIApplication.m:4353)
17  Kodi Remote                   	0x100990c34 main + 80 (main.m:15)
18  libdyld.dylib                 	0x182ba18e0 start + 4 (:-1)
```

## Summary for release notes
<!--- This will be shown in Testflight to end users / testers -->
<!--- Please fill in, as usually PR title isn't clear to ordinary users -->
<!--- You can keep it empty if it's identical to the PR title though -->
<!--- If your changes don't modify app files, please add prefix [not app] -->
Bugfix: Avoid potential crash in Global Search
Maintenance: Rework handling of active tab with focus on the library view

kambala-decapitator commented on XBMC Remote/DetailViewController.m:
```diff
@@ -699,39 +699,54 @@ - (NSString*)getGlobalSearchThumb:(NSDictionary*)item {
-            lookup = AppDelegate.instance.globalSearchMenuLookup[GLOBALSEARCH_INDEX_TVSHOWS];
-        }
+        NSArray *lookup = [self getGlobalSearchLookup:item];
         if (lookup) {
```
more correct would be to check number of items in array

wutschel replied:
Good point, will do.

kambala-decapitator commented on XBMC Remote/DetailViewController.m:
```diff
@@ -699,39 +699,54 @@ - (NSString*)getGlobalSearchThumb:(NSDictionary*)item {
+    int activeTab = choosedTab;
+    if (globalSearchView) {
+        NSArray *lookup = [self getGlobalSearchLookup:item];
+        if (lookup) {
```
here as well

wutschel replied:
agreed

kambala-decapitator commented on XBMC Remote/DetailViewController.m:
```diff
@@ -5995,12 +6006,9 @@ - (void)viewDidLoad {
-    if (menuItem.chooseTab) {
-        choosedTab = menuItem.chooseTab;
-    }
+    choosedTab = menuItem.chooseTab ?: 0;
```
not sure why you need ternary operator here - the "false" branch would trigger only when the value is 0

wutschel replied:
True, will just reduce to `choosedTab = menuItem.chooseTab`.

kambala-decapitator reviewed (approved):
it's time to write that word without mistakes :) **chosen** instead of **choosed**

and please fix conflicts

wutschel commented:
> it's time to write that word without mistakes :) **chosen** instead of **choosed**

This is also catching my eye each time I am seeing it. Will add a commit for this on top of all of this.

wutschel commented:
> > it's time to write that word without mistakes :) **chosen** instead of **choosed**
> 
> This is also catching my eye each time I am seeing it. Will add a commit for this on top of all of this.

Rebased, squashed and also made the grammar change from `choosedTab` to `chosenTab`.
